web/packages/teleport/src/Discover/AwsMangementConsole/CreateAppAccess/CreateAppAccess.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Box, Text, Flex, Link } from 'design';
+import TextEditor from 'shared/components/TextEditor';
+import { Danger } from 'design/Alert';
+import { ToolTipInfo } from 'shared/components/ToolTip';
+import { useAsync } from 'shared/hooks/useAsync';
+
+import { TextSelectCopyMulti } from 'teleport/components/TextSelectCopy';
+import { useDiscover } from 'teleport/Discover/useDiscover';
+import { integrationService } from 'teleport/services/integrations';
+import cfg from 'teleport/config';
+import { Container } from 'teleport/Discover/Shared/CommandBox';
+import { splitAwsIamArn } from 'teleport/services/integrations/aws';
+
+import { ActionButtons, Header, Mark } from '../../Shared';
+
+import { AppCreatedDialog } from './AppCreatedDialog';
+
+const IAM_POLICY_NAME = 'AWSAppAccess';
+
+export function CreateAppAccess() {
+  const { agentMeta, updateAgentMeta, emitErrorEvent, nextStep } =
+    useDiscover();
+  const { awsIntegration } = agentMeta;
+
+  const [attempt, createApp] = useAsync(async () =>
```

Comment:
Nit: Since we have an `async` function, do we want to switch promises to `await` syntax?